app/validators/array_hostnames_validator.rb

```diff
@@ -0,0 +1,6 @@
+class ArrayHostnamesValidator < ActiveModel::EachValidator
+  def validate_each(record, attribute, value)
+    regexp = /^(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]*[a-zA-Z0-9])\.)*([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9\-]*[A-Za-z0-9])$/
```

Comment:
I'd prefer using something built-in, how about [URI.regexp](http://ruby-doc.org/stdlib-1.9.3/libdoc/uri/rdoc/URI.html#method-c-regexp)